src/Platform/Microsoft.Testing.Platform.MSBuild/Tasks/TestingPlatformEntryPointTask.cs

```diff
@@ -54,82 +57,89 @@ public override bool Execute()
         taskLoggingHelper.LogMessage(MessageImportance.Normal, $"Entrypoint source:\n'{entryPointSource}'");
         fileSystem.WriteAllText(testingPlatformEntryPointSourcePath.ItemSpec, entryPointSource);
     }
 
-    private static string GetEntryPointSourceCode(string language)
+    private static string GetEntryPointSourceCode(string language, string rootNamespace)
     {
         if (language == CSharpLanguageSymbol)
         {
-            return """
+            return $$"""
 //------------------------------------------------------------------------------
 // <auto-generated>
 //     This code was generated by Microsoft.Testing.Platform.MSBuild
 // </auto-generated>
 //------------------------------------------------------------------------------
 
-[global::System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
-internal sealed class TestingPlatformEntryPoint
+namespace {{rootNamespace}}
 {
-    public static async global::System.Threading.Tasks.Task<int> Main(string[] args)
+    [global::System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
+    internal sealed class TestingPlatformEntryPoint
     {
-        global::Microsoft.Testing.Platform.Builder.ITestApplicationBuilder builder = await global::Microsoft.Testing.Platform.Builder.TestApplication.CreateBuilderAsync(args);
-        SelfRegisteredExtensions.AddSelfRegisteredExtensions(builder, args);
-        using (global::Microsoft.Testing.Platform.Builder.ITestApplication app = await builder.BuildAsync())
+        public static async global::System.Threading.Tasks.Task<int> Main(string[] args)
         {
-            return await app.RunAsync();
+            global::Microsoft.Testing.Platform.Builder.ITestApplicationBuilder builder = await global::Microsoft.Testing.Platform.Builder.TestApplication.CreateBuilderAsync(args);
+            SelfRegisteredExtensions.AddSelfRegisteredExtensions(builder, args);
+            using (global::Microsoft.Testing.Platform.Builder.ITestApplication app = await builder.BuildAsync())
+            {
+                return await app.RunAsync();
+            }
         }
     }
 }
 """;
         }
         else if (language == VBLanguageSymbol)
         {
-            return """
+            return $$"""
 '------------------------------------------------------------------------------
 ' <auto-generated>
 '     This code was generated by Microsoft.Testing.Platform.MSBuild
 ' </auto-generated>
 '------------------------------------------------------------------------------
 
-<System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage>
-Module TestingPlatformEntryPoint
+Namespace {{rootNamespace}}
+    <System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage>
+    Module TestingPlatformEntryPoint
 
-    Function Main(args As Global.System.String()) As Global.System.Int32
-        Return MainAsync(args).Result
-    End Function
+        Function Main(args As Global.System.String()) As Global.System.Int32
+            Return MainAsync(args).Result
```

Comment:
Unrelated to your changes, but this potentially doesn't match the behavior of C#. It may be worth updating this to `GetAwaiter().GetResult()` instead to match C#. And maybe have some test (at least manual) to verify that.

Do you want us to track it in an issue? Or was this an explicit decision for some reason?

Reply:
No idea. Let's see if fixing it causes any issue.